DietPi-Banner | add letsencrypt certificate details

gary2002 opened the pull request:
Addition to check if a letsencrypt certificate exists and if so, print the certificate expiration details.

This can act as a simple check to see if letsencrypt certificates are updating over time.

<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->

gary2002 committed:
banner with letsencrypt certificate details

Addition to check if a letsencrypt certificate exists and if so, print the certificate expiration details.

This can act as a simple check to see if letsencrypt certificates are updating over time.

gary2002 committed:
Update dietpi-banner

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -56,15 +56,16 @@
 	if (( $G_HW_MODEL == 20 ))
 	then
-		aENABLED=(1 0 0 0 0 1 0 1 0 0 0 1 1 0 0 1)
+		aENABLED=(1 0 0 0 0 1 0 1 0 0 0 1 1 0 0 1 1)
```
This should not be enabled by default:
```suggestion
		aENABLED=(1 0 0 0 0 1 0 1 0 0 0 1 1 0 0 1 0)
```

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -56,15 +56,16 @@
+		aENABLED=(1 0 0 0 0 1 0 1 0 0 0 1 1 0 0 1 1)
 	else
-		aENABLED=(1 0 1 0 0 1 0 0 0 0 0 1 1 0 0 1)
+		aENABLED=(1 0 1 0 0 1 0 0 0 0 0 1 1 0 0 1 1)
```
```suggestion
		aENABLED=(1 0 1 0 0 1 0 0 0 0 0 1 1 0 0 1 0)
```

gary2002 replied:
Happy with that. Gary

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -56,15 +56,16 @@
 		'VPN status'
 		'Large hostname'
 		'Print credits'
+		'Certificate Valid to'
```
Since this is what is shown in the menu, I think it should me more meaningful:
```suggestion
		'Let'\''s Encrypt cert status'
```

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -263,6 +264,18 @@ $GREEN_LINE"
+				break
+			done
+		fi
+		(( ${aENABLED[16]} == 1 )) && echo -e "$GREEN_BULLET ${aCOLOUR[1]}${aDESCRIPTION[16]} $GREEN_SEPARATOR $certinfo"
```
Suggested change:
- Do not obtain cert info if the banner option is disabled
- Do not assign a variable which is called only a single time
- But declare `i` as local variable
- Instead of checking whether the parent directory exist, check whether the certificate itself exists
```suggestion
		# Let's Encrypt cert status
		if (( ${aENABLED[16]} == 1 ))
		then
			local i certinfo='No certificate found'
			for i in /etc/letsencrypt/live/*/cert.pem
			do
				[[ -f $i ]] || continue
				certinfo=$(openssl x509 --enddate --noout --in "$i" | mawk '/notAfter=/{print "Valid until " substr($2,1) "-" substr($1,10) "-" substr($4,1) " @ " substr($3,1)}' 2>&1)
				break
			done
			echo -e "$GREEN_BULLET ${aCOLOUR[1]}${aDESCRIPTION[16]} $GREEN_SEPARATOR $certinfo"
		fi
```

gary2002 committed:
Update dietpi/func/dietpi-banner

Happy with that.

Co-authored-by: MichaIng <micha@dietpi.com>

Joulinar committed:
v8.17

DietPi-Banner | add letsencrypt certificate details, do not enable function by default

Co-authored-by: MichaIng <micha@dietpi.com>

Joulinar committed:
v8.17

DietPi-Banner | add letsencrypt certificate details, change menu name

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng committed:
Do not enable Let's Encrypt status by default on VMs

MichaIng committed:
v8.17

- CHANGELOG | DietPi-Banner: Added a new option to show the Let's Encrypt certificate status (expiry date), when installed via dietpi-letsencrypt or Certbot. Many thanks to @gary2002 for implementing this option.

MichaIng committed:
v8.17

- DietPi-Banner | Simplify awk for obtaining the certificate expiry date

MichaIng committed:
v8.17

- DietPi-Banner | Do not obtain IP info if it is not printed anyway
- DietPi-Banner | Let's Encrypt cert status: Print in format "YYYY-MMM-DD HH:MM:SS"

MichaIng committed:
v8.17

- CHANGELOG | DietPi-Banner: Add link to PR
